pkg/skaffold/schema/versions_test.go

```diff
@@ -472,3 +472,27 @@ func TestCantUpgradeFromLatestVersion(t *testing.T) {
 	_, err := factory().Upgrade()
 	testutil.CheckError(t, true, err)
 }
+
+func TestParseConfigAndUpgradeToUnknownVersion(t *testing.T) {
+	testutil.Run(t, "", func(t *testutil.T) {
+		t.NewTempDir().
+			Write("skaffold.yaml", fmt.Sprintf("apiVersion: %s\nkind: Config\n%s", latest.Version, minimalConfig)).
+			Chdir()
+
+		_, err := ParseConfigAndUpgrade("skaffold.yaml", "unknown")
+
+		t.CheckErrorContains(`unknown api version: "unknown"`, err)
+	})
+}
+
+func TestParseConfigAndUpgradeToOlderVersion(t *testing.T) {
```

Comment:
maybe i misunderstood. We would to provide an ability to fix skaffold config to version older than the current latest right?
e.g. if skaffold-v1.2, the config version shipped with it is `skaffold/v1alpha2`. However users want to be `skaffold/v1`
did i get it right?

Reply:
yes, they can choose a version that's older than `latest` but not a version that's older than the version used in the skaffold.yaml. That would be a downgrade.

Reply:
hmm that would mean, if a user
1. Upgrades the version to latest using `skaffold fix` and then realizes, they updated the config to `skaffold/v1aplha2`. 
2. Now they want to go back to `skaffold/v1` version since they don't want to rely on `alpha` config. 
It is currently not possible right because `skaffold/v1alpha2` is greater than `skaffold/v1`

Reply:
i think for now, i am fine with this. But in practice to come out of this situation, users will have to
1. revert back to old config, 
2. try `skaffold fix --version`

Reply:
indeed. Supporting downgrades would be a lot of work